src/builtins.c

```diff
@@ -454,6 +454,17 @@ JL_CALLABLE(jl_f__apply_pure)
     return ret;
 }
 
+// this is like `_apply`, but always runs in the newest world
+JL_CALLABLE(jl_f__apply_latest)
+{
+    jl_ptls_t ptls = jl_get_ptls_states();
+    size_t last_age = ptls->world_age;
+    ptls->world_age = jl_world_counter;
```

Comment:
It's mostly just by-definition. A pure function is one for which the world-counter has no effect. In those other examples, those two functions try to change `jl_world_counter`, so they have to throw an error. In this example, it's only trying to read `jl_world_counter`, which is OK, as long as it can't see the value change.